lib/rack/attack/throttle.rb

```diff
@@ -22,20 +22,21 @@ def cache
         Rack::Attack.cache
       end
 
-      def matched_by?(request)
+      def matched_by?(request, use_offset = false)
         discriminator = discriminator_for(request)
         return false unless discriminator
 
         current_period  = period_for(request)
         current_limit   = limit_for(request)
-        count           = cache.count("#{name}:#{discriminator}", current_period)
+        count           = cache.count("#{name}:#{discriminator}", current_period, use_offset)
 
         data = {
           discriminator: discriminator,
           count: count,
           period: current_period,
           limit: current_limit,
-          epoch_time: cache.last_epoch_time
+          epoch_time: cache.last_epoch_time,
+          retry_after: cache.last_retry_after_time
```

Comment:
I think this will set the `retry_after` value on every request (not just the throttled ones). Is that the intended behavior? or should it be only for throttled requests?